### ISSUE TITLE
--psyco speedup option for trial, for everything else too

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7379]
|<img alt="ne1uno's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ne1uno reported|
|-|-|
|Trac ID|trac#145|
|Type|enhancement|
|Created|2003-07-29 09:48:43Z|

<details><summary>Searchable metadata</summary>

```
trac-id__145 145
type__enhancement enhancement
reporter__ne1uno ne1uno
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1059472123000000 1059472123000000
changetime__1475245475991716 1475245475991716
version__ 
owner__ 
cc__ne1uno cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ne1uno's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ne1uno commented|
|-|-|

```
#!html
<pre>
feature wish --psyco , needs to be integrated internally 
since the  effect
is on a per file per class basis. this would require some 
confusing
try excepts and passing  command line switch parsing 
throughout the import chain.
there was a mention in one of the pycon or rant papers I 
forget from where
mentioning  a substantial speedup in several important 
measurable parameters. (i386 only as yet)
obviously this could be enhanced  with a profile of the 
most used classes,
so maybe buildbot can be trained to run some examples with 
and without
to develop the necessary data to determine where to apply 
the most improvements.
not sure adding it to only your own code is usefull.

I don't entirely trust the trial time reporting.
run twice in a row the 2nd time says its about half the 
time but on the wall clock its about the same 11 minutes
even though the 2nd time all the test files didnt have to 
be recreated. maybe another issue there. P233 win98

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
slyphon can have a look at this, since it is mostly trial
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
umm, well, the --psyco option *may* go into trial (i'd considered this and
tested it briefly in retrial), but i'd reccomend against running it on buildbot,
as it can segfault, and does regularly when running the twisted.test.test_pb
tests because of some kind of weirdness in a __del__.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
I am closing this issue. I did some benchmarks using psyco, and not only is it
unstable in running the twisted tests (with trial -vR twisted) but in my tests,
it actually causes the tests to run slower.

The tests in retrial already run 3-4x faster (consistently) than with the
current version of trial that's in trunk, so i doubt there is much more to be
gained through the use of psyco.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_6"></a>|@markrwilliams commented|
|-|-|

In changeset 63f3e6e5acfc9042cb0a7c925ed321d3814d3241
```
#!CommitTicketReference repository="" revision="63f3e6e5acfc9042cb0a7c925ed321d3814d3241"
Merge pull request #145 from twisted/timeout-deferreds-5786-5

Merge twisted/timeout-deferreds-5786-5: addTimeout function on defer.Deferred

Author: cyli 
Reviewer: adiroiban, theisencouple, manishtomar, markrwilliams
Fixes: #5786

Added twisted.internet.defer.Deferred.addTimeout method to enable timeouts of deferreds.
```
